validator/sawtooth_validator/consensus/proxy.py

```diff
@@ -187,10 +201,9 @@ def state_get(self, block_id, addresses):
         return result
 
     def _get_blocks(self, block_ids):
-        try:
-            return [
-                self._block_cache[block_id.hex()]
-                for block_id in block_ids
-            ]
-        except KeyError:
-            raise UnknownBlock()
+        for block_id in block_ids:
+            if block_id.hex() not in self._block_manager:
+                raise UnknownBlock()
+        return [b for b in self._block_manager.get([
+            block_id.hex() for block_id in block_ids
+        ])]
```

Comment:
I think this could be done more efficiently by constructing the list to return first, and then checking its length against the length of the input block_ids to determine if a block was missing.